client/homebrew/brewRenderer/brewRenderer.jsx

```diff
@@ -76,9 +75,80 @@ const BrewRenderer = (props)=>{
 		rawPages = props.text.split(/^\\page$/gm);
 	}
 
-	useEffect(()=>{ // Unmounting steps
-		return ()=>{window.removeEventListener('resize', updateSize);};
+	useEffect(() => {
+		const locationHash = window.location.hash;
+		const iframe = document.getElementById('BrewRenderer');
+		
+		const pageIdRegex = /^#p\d+$/;
+		
+		const handleIframeLoad = () => {
+			setTimeout(() => {
+				if (pageIdRegex.test(locationHash)) {
+					const pageNumber = parseInt(locationHash.slice(2));
+					scrollToPage(pageNumber - 1);
+				} else {
+					const elementId = locationHash.slice(1);
+					if (elementId) {
+						getPageContainingElement(elementId)
+							.then((pageNumber) => {
+								if (pageNumber !== -1) {
+									scrollToPage(pageNumber);
+								}
+							})
+							.catch((error) => {
+								console.error('Error:', error);
+							});
+					}
+				}
+			}, 100);
+		};
+		
+		if (locationHash) {
+			iframe.addEventListener('load', handleIframeLoad);
+		}
+
+		return () => {
+			if (locationHash) {
+				iframe.removeEventListener('load', handleIframeLoad);
+			}
+			window.removeEventListener('resize', updateSize);
+		};
 	}, []);
+	
+
+	const scrollToPage = (pageNumber) => {
+		const iframe = document.getElementById('BrewRenderer');
+        if (iframe && iframe.contentWindow) {
+            const brewRenderer =
+                iframe.contentWindow.document.querySelector('.brewRenderer');
+            if (brewRenderer) {
+                const pages = brewRenderer.querySelectorAll('.page');
+                if (pageNumber > pages.length) {
+                    console.log('page not found');
+                } else {
+                    pages[pageNumber].scrollIntoView({ block: 'start' });
+                }
+            }
+        }
+    };
+	
+	const getPageContainingElement = (elementId) => {
```

Comment:
If we really do want to get the parent page of the element, I believe we can just use:

`document.getElementById("theId").closest(".page")`

`.closest` grabs the nearest ancestor that matches a certain query.